Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnInit.hs

```diff
@@ -67,6 +72,7 @@ onInit req = do
   riderPhoneCountryCode <- decRider.mobileCountryCode & fromMaybeM (PersonFieldNotPresent "mobileCountryCode")
   riderPhoneNumber <- decRider.mobileNumber & fromMaybeM (PersonFieldNotPresent "mobileNumber")
   bppBookingId <- booking.bppBookingId & fromMaybeM (BookingFieldNotPresent "bppBookingId")
+  bapUrl <- asks (.nwAddress) <&> #baseUrlPath %~ (<> "/cab/v1/" <> T.unpack merchant.id.getId)
```

Comment:
This code is the same for all apis, can be reused. Also can be moved on ACL level (where it was earlier). You can pass merchant in `req` instead of three fields `city`, `bapId`, `bapUrl`

Reply:
+1